misc/experimental/promise/api/celix/PromiseFactory.h

```diff
@@ -0,0 +1,50 @@
+/**
+ *Licensed to the Apache Software Foundation (ASF) under one
+ *or more contributor license agreements.  See the NOTICE file
+ *distributed with this work for additional information
+ *regarding copyright ownership.  The ASF licenses this file
+ *to you under the Apache License, Version 2.0 (the
+ *"License"); you may not use this file except in compliance
+ *with the License.  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *Unless required by applicable law or agreed to in writing,
+ *software distributed under the License is distributed on an
+ *"AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ *specific language governing permissions and limitations
+ *under the License.
+ */
+
+#pragma once
+
+#include "celix/Deferred.h"
+
+namespace celix {
+
+
+    class PromiseFactory {
```

Comment:
Adding a `resolved` and `failed` factory methods should be relatively easy, and increases usability in areas where the result is not calculated asynchronously.

Reply:
I will look into this and see if I can update this already in PR. 
Else this will be done in a separate PR (I expect still a few more PRs, before this implementation can be considered complete).

Reply:
added